test-data/unit/check-overloading.test

```diff
@@ -787,7 +787,11 @@ class D(A):
     def f(self, x: str) -> str: return ''
 [out]
 tmp/foo.pyi:12: error: Signature of "f" incompatible with supertype "A"
+tmp/foo.pyi:12: note: Signature of "f" in superclass: "Overload(def f(self, x: builtins.int) -> builtins.int, def f(self, x: builtins.str) -> builtins.str)"
```

Comment:
It's fine to repeat `note` multiple times. Some tools might expect the path and the line number to be always present on every line. If there are very many overload items (say, more than 3) we can just show the first 3 and something like (<additional 3 overload variants omitted>).